### ISSUE TITLE
Support `preserve_counts` in job plan operations

### ISSUE THREAD

**Issue by lgfa29** [hashicorp/nomad#19845]
### Proposal

Job scaling operations can modify a task group count value without changing the jobspec itself. To avoid having to keep job spec files in sync with scaling operations, https://github.com/hashicorp/nomad/issues/8158 introduced a new option, `PreserveCounts` to job registration. When set to `true`, Nomad will ignore the `count` values passed in the input job.

But this option is not supported in the job plan operation, so job plans still report a change that may not happen.

### Use-cases

Consistent result between `nomad job plan` and `nomad job run -preserve-counts`.

### Attempted Solutions

Ignore plan result.

**Comment by Jamesits** [comment]
Another possibility: define `count` as the job count on only the first deployment (or job creation). After a job has been created, make the real job count a part of the job state rather than the definition. 

This way we might also don't need to create a new version every time we want to scale out/in a job...

**Comment by lgfa29** [comment]
Yeah, there's been some discussion about splitting a job between it's "spec" and "dynamic" components. 

But this issue is more about API parity. `POST /v1/jobs` has a [`PreserveCounts`](https://developer.hashicorp.com/nomad/api-docs/jobs#preservecounts) option so `POST /v1/job/:job_id/plan` should support this as well.

**Comment by EugenKon** [comment]
This is another issue, but it is related. Let's say for example we have 100 copies of the same job deployed. When we nomad plan/run it will go one by one: stop the old and create new jobs. This could take hours. If we have 100 copies there exists small redundancy for sure. So it could be better to deploy new version of job by groups, eg. shutdown 10 jobs and create new 10. Or, if we relay on the minimum amount of jobs, we can shutdown/create that amount. After this having minimum amount of new services it could be safe to shutdown the rest and create new one. Or continue to use group strategy: shutdown X and start X new one.
This will really speedup things.

**Comment by tgross** [comment]
I think you need to adjust your [`update.max_parallel`](https://developer.hashicorp.com/nomad/docs/job-specification/update#max_parallel), which defaults to 1.

**Comment by EugenKon** [comment]
@tgross How about `max_parallel=15%`? This could be more handy for big clusters with 1000 EC2 instances/job instances.
And I am thinking about `slow_start=2` which means start only two new jobs and check they are healthy before Nomad can start another 1000*0.15 jobs, otherwise there is no reason to start a bunch of 150 jobs if they are broken (even 2 of them can not start properly).
